### PR TITLE
OSX_arm64 support

### PR DESCRIPTION
**Pull request by pfebrer**
Modification of https://github.com/conda-forge/siesta-feedstock/pull/52 to try to make cross-compilation work.

### CODE REVIEW

**Comment by pfebrer** [comment]
Cool, so I managed to overcome the tests problem. Now it gets to the build phase, but it fails because of a wrong flag for the fortran compiler:

```
[  0%] Building Fortran object _deps/libfdf-build/CMakeFiles/libfdf-lib.dir/src/prec.F90.o
cd $SRC_DIR/obj_cmake/_deps/libfdf-build && $PREFIX/bin/mpifort  -I$SRC_DIR/obj_cmake/_deps/libfdf-build/include -march=armv8.3-a -ftree-vectorize -fPIC -fno-stack-protector -O2 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/siesta-5.0.0rc1 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -I$PREFIX/lib -fallow-argument-mismatch -O3 -march=native -mmacosx-version-min=11.0 -Jinclude -fPIC -c $SRC_DIR/obj_cmake/_deps/libfdf-src/src/prec.F90 -o CMakeFiles/libfdf-lib.dir/src/prec.F90.o
f951: Error: unknown value ‘native’ for ‘-march’
f951: note: valid arguments are: armv8-a armv8.1-a armv8.2-a armv8.3-a armv8.4-a armv8.5-a armv8.6-a armv8.7-a armv8.8-a armv8-r armv9-a
make[2]: *** [_deps/libfdf-build/CMakeFiles/libfdf-lib.dir/build.make:130: _deps/libfdf-build/CMakeFiles/libfdf-lib.dir/src/prec.F90.o] Error 1
make[2]: Leaving directory '$SRC_DIR/obj_cmake'
make[1]: *** [CMakeFiles/Makefile2:2494: _deps/libfdf-build/CMakeFiles/libfdf-lib.dir/all] Error 2
make[1]: Leaving directory '$SRC_DIR/obj_cmake'
make: *** [Makefile:149: all] Error 2
```

It looks like it is correctly set to `-march=armv8.3-a` but then overwritten by `-march=native`. I don't know where this flags come from.

**Comment by pfebrer** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -5,4 +5,4 @@ mpi:
 
 channel_targets:
   # Remove this for proper releases
-  - conda-forge siesta_rc
```
Comment (zerothi) [inline recipe/conda_build_config.yaml:8]: this should stay until 5.0.0 is properly released.

**Review by zerothi — CHANGES_REQUESTED** [review]
There are some more problems that needs fixed, but this is more upstream in the Siesta code.

I'll open an issue to track these details

**Comment by pfebrer** [comment]
The no mpi version seems to work. Nice!!! Let's see if your MPI changes are enough to fix the other ones :)

**Comment by zerothi** [comment]
> The no mpi version seems to work. Nice!!! Let's see if your MPI changes are enough to fix the other ones :)

That's good, then I would be much more confident it works, you could try and take out the commit (from my PR) as a patch and see?
I only marked it as a draft because I don't know if more needs to be done.

**Comment by pfebrer** [comment]
Which kinds should I use for arm_64? Should I use your example on the PR:

```
-DSIESTA_REAL_KINDS="4;8"
-DSIESTA_INTEGER_KINDS="4;8"
```

?

**Comment by zerothi** [comment]
> Which kinds should I use for arm_64? Should I use your example on the PR:
> 
> ```
> -DSIESTA_REAL_KINDS="4;8"
> -DSIESTA_INTEGER_KINDS="4;8"
> ```
> 
> ?

It is compiler specific, not architechture. So I think those should suffice for gcc. Largely all compilers use these values, a few don't... :)

**Comment by zerothi** [comment]
the patch looks a bit weird... I guess you did `git diff`? One should do `git diff --patch <>` then the message etc. won't stay behind.

**Comment by pfebrer** [comment]
I use gitlab's graphical interface to get the patch. I don't know how to get the patch easily, since it must patch 5.0.0-beta1 and the MR is against `master`.

**Comment by pfebrer** [comment]
Yeah I don't know how to apply those changes :sweat_smile: So maybe we can try with the next release.

**Comment by zerothi** [comment]
@conda-forge-admin, please rerender

**Comment by zerothi** [comment]
@pfebrer can you check whether this works on arm?

**Comment by zerothi** [comment]
haha, its just annoying one cannot check it ;)

**Pull request merged**